System.IO.FileSystem/File.cs

```diff
@@ -45,8 +45,16 @@ public static class File
             string destFileName,
             bool overwrite)
         {
-            // TODO: File Handling missing
-            
+            if (string.IsNullOrEmpty(sourceFileName))
```

Comment:
So far we've been avoiding these "verbose" exceptions. It costs a lot of flash space to store all these strings...
I understand that's convenient, but the list of the possible exceptions on the Intellisense comments and the developer looking closer to figure out what is missing is the way to go.

Also this is checked at the native code when the argument is to be used and an exception is thrown, so the information bubbling up is accurate. Maybe not complete, but it's there.